Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -69,51 +140,187 @@ def __init__(self, initial=None, f=None, g=ZeroFunction(), **kwargs):
             else:
                 raise ValueError('{} received both initial and the deprecated x_init parameter. It is not clear which one we should use.'\
                     .format(self.__class__.__name__))
-        
-        if initial is not None and f is not None:
-            self.set_up(initial=initial, f=f, g=g)
 
-    def set_up(self, initial, f, g=ZeroFunction()):
-        '''initialisation of the algorithm
+        self._step_size = None
+
+        # set up FISTA
+        self.set_up(initial=initial, f=f, g=g, step_size=step_size, **kwargs)
+
+    def set_up(self, initial, f, g, step_size, **kwargs):
+        """ Set up of the algorithm
+        """
+
+        self.initial = initial
+        self.f = f
+        self.g = g
+
+        # set step_size
+        self._set_step_size(step_size=step_size)
 
-        :param initial: Initial guess ( Default initial = 0)
-        :param f: Differentiable function
-        :param g: Convex function with " simple " proximal operator'''
+        # check convergence criterion for FISTA is satisfied
+        if kwargs.get('check_convergence_criterion', True):
+            self._check_convergence_criterion()
 
         print("{} setting up".format(self.__class__.__name__, ))
-        
+
         self.y = initial.copy()
         self.x_old = initial.copy()
         self.x = initial.copy()
         self.u = initial.copy()
 
-        self.f = f
-        self.g = g
-        if f.L is None:
-            raise ValueError('Error: Fidelity Function\'s Lipschitz constant is set to None')
-        self.invL = 1/f.L
         self.t = 1
         self.configured = True
+
         print("{} configured".format(self.__class__.__name__, ))
 
-            
+
     def update(self):
+
+        r"""Performs a single iteration of FISTA
+
+        .. math::
+
+            \begin{cases}
+                x_{k+1} = \mathrm{prox}_{\alpha g}(x_{k} - \alpha\nabla f(x_{k}))\\
```

Comment:
the right hand side should have y_k, instead of x_k, per https://www.math.mcgill.ca/yyang/comp/papers/FISTA.pdf, right?

Reply:
Fixed